src/runtime/node/cluster/index.ts

```diff
@@ -0,0 +1,52 @@
+// Reference: https://github.com/nodejs/node/blob/main/lib/internal/cluster/primary.js
+import noop from "../../mock/noop";
+import mock from "../../mock/proxy";
+import type cluster from "node:cluster";
+import type { Cluster, Worker as _Worker } from "node:cluster";
+import { EventEmitter } from "../events";
+
+// A mapped type used internally to allow assigning to readonly fields like `isPrimary`
+type MutableCluster = {
+  -readonly [key in keyof Cluster]: Cluster[key];
+} & {
+  Worker: typeof _Worker;
+};
+
+// eslint-disable-next-line unicorn/prefer-event-target
+const _cluster = new EventEmitter() as MutableCluster;
+
+export const disconnect: typeof cluster.disconnect = noop;
+export const fork: typeof cluster.fork = () =>
+  mock.__createMock__("cluster.Worker");
+export const isMaster: typeof cluster.isMaster = true;
+export const isPrimary: typeof cluster.isPrimary = true;
+export const isWorker: typeof cluster.isWorker = false;
+export const SCHED_NONE: typeof cluster.SCHED_NONE = 1;
+export const SCHED_RR: typeof cluster.SCHED_RR = 2;
+export const schedulingPolicy: typeof cluster.schedulingPolicy = SCHED_RR;
+export const settings: typeof cluster.settings = {};
+export const setupPrimary: typeof cluster.setupPrimary = noop;
+export const setupMaster: typeof cluster.setupMaster = noop;
+export const workers: typeof cluster.workers = {};
+export const Worker: typeof _Worker = mock.__createMock__("cluster.Worker");
+
+// These 3 _functions don't exist on the EventEmitter type
+export const _events = (_cluster as any)._events;
+export const _eventsCount = (_cluster as any)._eventsCount;
+export const _maxListeners = (_cluster as any)._maxListeners;
```

Comment:
I think only named export of this from `node:cluster` is the `_cluster` instance do we need additional exports?

Reply:
according to our node API surface scans all of these are needed: https://workers-nodejs-compat-matrix.pages.dev/

Reply:
Strange but fair enough I can confirm with local Node.js (ESM) as well they get converted to named exports. Technically they are keys of default export...